Project/Source/Components/ComponentParticleSystem.cpp

```diff
@@ -0,0 +1,325 @@
+		}
+
+		ImGui::ColorEdit4("InitColor##", initC.ptr());
+		ImGui::ColorEdit4("FinalColor##", finalC.ptr());
+	}
+}
+
+//TODO: DINAMIC PARTICLE NOT HARCODED
+float3 ComponentParticleSystem::CreateVelocity() {
+	float x, y, z;
+	if (emitterType == EmitterType::CONE) {
+		ComponentTransform* transform = GetOwner().GetComponent<ComponentTransform>();
+		x = (float(rand()) / float((RAND_MAX)) * 0.2) - 0.2;
+		y = (float(rand()) / float((RAND_MAX)) * 0.5) - 0.0;
+		z = (float(rand()) / float((RAND_MAX)) * 0.5) - 0.2;
+		if (!randomDirection) return float3(0, velocity, 0);
+		return float3(x, y, z);
+	}
+	//TODO: DINAMIC PARTICLE NOT HARCODED
+	if (emitterType == EmitterType::SPHERE) {
+		ComponentTransform* transform = GetOwner().GetComponent<ComponentTransform>();
+		x = (transform->GetGlobalPosition().x) + (float(rand()) / float((RAND_MAX)) * 2.0) - 1.0f;
+		y = (transform->GetGlobalPosition().y) + (float(rand()) / float((RAND_MAX)) * 2.0) - 1.0f;
+		z = (transform->GetGlobalPosition().z) + (float(rand()) / float((RAND_MAX)) * 2.0) - 1.0f;
+		return float3(x, y, z);
+	}
+};
+
+float3 ComponentParticleSystem::CreatePosition() {
+	//TODO: DINAMIC PARTICLE NOT HARCODED
+	float x, y, z;
+
+	if (emitterType == EmitterType::CONE) {
+		ComponentTransform* transform = GetOwner().GetComponent<ComponentTransform>();
+		x = (transform->GetGlobalPosition().x);
+		z = (transform->GetGlobalPosition().z);
+		y = (transform->GetGlobalPosition().y);
+		return float3(x, y, z);
+	}
+	//TODO: DINAMIC PARTICLE NOT HARCODED
+	if (emitterType == EmitterType::SPHERE) {
+		ComponentTransform* transform = GetOwner().GetComponent<ComponentTransform>();
+		x = (transform->GetGlobalPosition().x) + (float(rand()) / float((RAND_MAX)) * 0.5) - 0.5;
+		z = (transform->GetGlobalPosition().z) + (float(rand()) / float((RAND_MAX)) * 0.5) - 0.5;
+		y = (transform->GetGlobalPosition().y) + (float(rand()) / float((RAND_MAX)) * 0.5) - 0.5;
+		return float3(transform->GetGlobalPosition());
+	}
+}
+
+void ComponentParticleSystem::CreateParticles(float nParticles, float vel) {
+	particles.Allocate(maxParticles);
+	for (Particle& currentParticle : particles) {
+		currentParticle.scale = float3(0.1f, 0.1f, 0.1f) * scale;
+		currentParticle.initialPosition = CreatePosition();
+		currentParticle.position = currentParticle.initialPosition;
+		currentParticle.direction = CreateVelocity();
+		currentParticle.velocity = vel;
+	}
+}
+
+void ComponentParticleSystem::Load(JsonValue jComponent) {
+	shaderID = jComponent[JSON_TAG_TEXTURE_SHADERID];
+
+	if (shaderID != 0) {
+		App->resources->IncreaseReferenceCount(shaderID);
+	}
+
+	textureID = jComponent[JSON_TAG_TEXTURE_TEXTUREID];
+
+	if (textureID != 0) {
+		App->resources->IncreaseReferenceCount(textureID);
+	}
+
+	CreateParticles(maxParticles, velocity);
+}
+
+void ComponentParticleSystem::Save(JsonValue jComponent) const {
+	jComponent[JSON_TAG_TEXTURE_SHADERID] = shaderID;
+	jComponent[JSON_TAG_TEXTURE_TEXTUREID] = textureID;
+}
```

Comment:
Particle settings are not being stored/loaded. If we create a scene and edit those particles, the changes will not be applied next time we run the scene (or on game)